docs/content/mongocxx-v3/api-abi-versioning.md

```diff
@@ -6,122 +6,252 @@ title = "API and ABI versioning"
+* API version number extensions are _not_ included in the real name (e.g. the `-alpha` in `1.2.3-alpha`).
+
+Examples of expected library real names include:
+
+```
+libbsoncxx.so.1.0.0
+libbsoncxx.so.2.3.4
+```
+
+CMake's `SOVERSION` target property and "namelink" behavior is used to also install symlinks to the library directory consistent with the Linux soname convention.
+These symlinks include the ABI version number as part of the library soname.
 
-## Parallel Header Installation
+For example, the following lib directory contains the expected library files for a bsoncxx library with API version `1.2.3` and ABI version `10`, where `a -> b` indicates `a` is a symlink to `b`:
 
-* For mongocxx, install all headers to `$PREFIX/mongocxx/v$ABI/`.
-* For bsoncxx, install all headers to `$PREFIX/bsoncxx/v$ABI/`.
-* We install a pkg-config file to shield consumers from this complexity.
+```
+${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/
+├── libbsoncxx.so.1.2.3
+├── libbsoncxx.so.10 -> libbsoncxx.so.1.2.3
+├── libbsoncxx.so -> libbsoncxx.so.10
+├── cmake/bsoncxx-1.2.3/
+│   ├── bsoncxx-config.cmake
+│   └── ...
+└── pkgconfig/
+    └── libbsoncxx.pc
+```
+
+The name of the CMake package for bsoncxx libraries is controlled by `BSONCXX_OUTPUT_BASENAME` (`MONGOCXX_OUTPUT_BASENAME` for mongocxx libraries).
+They should be imported according to CMake's [Config Mode Search Procedure](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure).
+For example, assuming the CXX Driver was installed to a prefix `$INSTALLDIR`:
+
+```bash
+# The CMake package may be imported via CMAKE_PREFIX_PATH (recommended)...
+cmake -S example -B example-build -D CMAKE_PREFIX_PATH="$INSTALLDIR"
+
+# Or via <PackageName>_ROOT (assuming BSONCXX_OUTPUT_NAME=bsoncxx)...
+cmake -S example -B example-build -D bsoncxx_ROOT="$INSTALLDIR"
+
+# Or via `PATHS` in the `find_package()` command... or etc.
+# find_package(bsoncxx CONFIG REQUIRED PATHS "$INSTALLDIR")
+```
 
-## Sonames and symlinks
+**NOTE:** The C Driver is a required dependency of CXX Driver. The CMake package config files for the C Driver may also need to be imported using the same (or similar) approach to satisfy this dependency.
 
-*Note that below examples are given for libmongocxx, but also apply to libbsoncxx*
+Although the CXX Driver also provides pkg-config files, users are strongly encouraged to use CMake package config files instead.
 
-*DSO = Dynamic Shared Object, to use Ulrich Drepper's terminology*
+The name of the pkg-config file for bsoncxx libraries is also controlled by `BSONCXX_OUTPUT_NAME` and follow the pattern `lib<basename>.pc`.
+Because the CXX Driver does not install pkg-config files to typical pkg-config system directories, the `PKG_CONFIG_PATH` environment variable may need to be used to augment pkg-config's search path, e.g.:
```

Comment:
```suggestion
The `PKG_CONFIG_PATH` environment variable may need to be used to augment pkg-config's search path, e.g.:
```

If the `CMAKE_INSTALL_PREFIX` override is disabled (with the recommended `MONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX=OFF`), I expect the package config file is in a default searched location:

```bash
cmake \
    -D CMAKE_CXX_STANDARD=17 \
    -D MONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX=OFF \
    -S. -B cmake-build
sudo cmake --build cmake-build --target install # Install /usr/local/lib/pkgconfig/libbsoncxx.pc
pkg-config --libs --cflags libbsoncxx # Locates libbsoncxx.pc.
```